eng/scripts/docs/tests/Service-Readme-Generation-Tests.ps1

```diff
@@ -0,0 +1,44 @@
+<#
+```
+Invoke-Pester -Output Detailed $PSScriptRoot/Service-Readme-Generation-Tests.ps1
+```
+#>
+
+Import-Module Pester
+
+BeforeAll {
+    . $PSScriptRoot/../Docs-ToC.ps1
+    . $PSScriptRoot/../../../common/scripts/logging.ps1
+}
+
+# Test plan:
+# 1. Tests on sucessfully building the onboarding package and its package info map.
+# 2. Tests on passing error csv file.
+# 3. Tests on missing metadata json file.
+Describe "Get-OnboardedDocsMsPackagesForMoniker" -Tag "UnitTest" {
+    # Passed cases
+    It "Build the map of onboarding package to package info" -TestCases @(
+        @{ DocRepoLocation = "$PSScriptRoot/inputs"; moniker="latest" }
+        @{ DocRepoLocation = "$PSScriptRoot/inputs"; moniker="preview" }
+    ) {
+        $onboardingPackages = Get-dotnet-OnboardedDocsMsPackagesForMoniker -DocRepoLocation $DocRepoLocation -moniker $moniker
+        foreach ($package in $onboardingPackages.GetEnumerator()) {
+            $package.Key | Should -Be $package.Value.Name
+            (ConvertTo-Json $package.Value) | Should -Be (Get-Content "$DocRepoLocation/metadata/$moniker/$($package.Key).json" -Raw)
```

Comment:
This could be fragile if the ordering or formatting changes. Pester doesn't have an obvious way to assert on whole PSCustomObjects or hash tables easily so this is probably the best use of time for now. If it gets fragile we'll need to reconsider.

Reply:
The value itself just convert from<->to Json. It is not hashmap, so the order will maintain.